### PR TITLE
Improved imports

### PR DESCRIPTION
**Pull request by markusdregi**
I'll start by removing the pylint-errors related to imports.

Afterwards I'm planning to look into explicit relative imports to stop submodules from inspecting the root module. This makes among others vscode cry.

https://github.com/microsoft/pyright/issues/439#issuecomment-569099772

### CODE REVIEW

File: ert3/engine/_record.py [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
-from ert3.engine import _utils
-
-from pathlib import Path
 import json
 from typing import Optional, TextIO
-import yaml
+from pathlib import Path
+
+import ert3
+from ert3.engine import _utils
```
Comment (jondequinor) [inline ert3/engine/_record.py:6]: Maybe I contradicted myself when I said we should keep the relative importing to the `__init__.py` files? Maybe it does make sense to use it everywhere, considering the new information from our Microsoft Fellow.

**Comment by jondequinor** [comment]
In `engine._run` there's a bunch of `stages_config` being thrown around, none are type hinted.

How will this work with the current import scheme?

File: ert3/engine/_record.py (hunk 1 above)
Comment (markusdregi) [reply ert3/engine/_record.py]: I'm thinking relative imports makes sense whenever things from the current (sub)module is imported. This is changed to a relative import in one of the later commits that I have now pushed...

**Comment by jondequinor** [comment]
A second problem is that we use `Record` and `RecordTransmitter` in the prefect ensemble. Hence we want to `import ert3`, but this will produce a cyclic import, since ert3 depends on `PrefectEnsemble`. We can fix this by moving these classes out of ert3. Or fix the import scheme, because it's not always straight forward to reason about it (i.e. importing one thing will import an arbitrary set of other things not strictly related to what you wanted in the first place).

**Comment by markusdregi** [comment]
> A second problem is that we use Record and RecordTransmitter in the prefect ensemble.

Yeah, I did see that... I think we need a proper discussion on project layout in general. But this PR is already opinionated, so it might be the wrong place to launch the discussion 🤔

**Comment by markusdregi** [comment]
> Pondering how will this work with the current import scheme, I guess all public symbols will need to be "exported". But then I think readability will take a huge hit if they need to be like ert3.config.StagesConfig.

I think currently in master they are passed around as `ert3.config.StagesConfig` when using type hinting. But there is nothing preventing one from doing `from ert3.config import StagesConfig` also...

**Comment by markusdregi** [comment]
@jondequinor: Created an issue #1520 and I hope we get to the above discussion also over time...

**Comment by markusdregi** [comment]
This is now widely outdated, the easily mergeable things have been extracted and lessons were learnt. If to be restarted we should just start over again in a fresh PR.

**Pull request closed without merging**